plugins/modules/vmware_vm_storage_policy.py

```diff
@@ -116,11 +116,36 @@
 from ansible.module_utils.basic import AnsibleModule
 from ansible_collections.community.vmware.plugins.module_utils.vmware_spbm import SPBM
 from ansible_collections.community.vmware.plugins.module_utils.vmware import vmware_argument_spec
+from ansible_collections.community.vmware.plugins.module_utils.vmware_rest_client import VmwareRestClient
+
+
+class TagCategoryRestClient(VmwareRestClient):
+    def __init__(self, module):
+        super(TagCategoryRestClient, self).__init__(module)
+        self.category_service = self.api_client.tagging.Category
+        self.tag_service = self.api_client.tagging.Tag
+
+    def check_category_exists(self):
+        for category in self.category_service.list():
+            category_obj = self.category_service.get(category)
+            if category_obj.name == self.params.get('tag_category'):
+                return True
+
+        return False
+
+    def check_tag_exists(self):
+        for tag in self.tag_service.list():
+            tag_obj = self.tag_service.get(tag)
+            if tag_obj.name == self.params.get('tag_name'):
```

Comment:
1. Can we move this method to `VmwareRestClient` class?
2. We can have multiple tags with the same name in different categories. Can you please add logic to check the given category as well?

Reply:
Thank you for reviewing.
okay, I’m going to work on it.

Reply:
In I rechecked an implementation of VmwareRestClient class, I was able to implement using already defined functions in VmwareRestClient class the if condition for a category or tag that doesn't exist.  
So I removed TagCategoryRestClient class.